pytentiostat/main.py

```diff
@@ -3,19 +3,65 @@
 from pytentiostat.reporter import save_data_to_file
 from pytentiostat.tester import experiment
 from pytentiostat.routines import startup_routine, closing_routine
+from pytentiostat.plotter import plot_updater, plot_initializer
+import matplotlib.pyplot as plt
+import sys
+import time
 
-config_data = parse_config_file()
 
-# Initialize Experiment
-com, board, a0, a2, d9 = startup_routine()
+class BoardCom:
+    def __init__(self):
+        self.com, self.board, self.pin_a0, self.pin_a2, self.pin_d9 = startup_routine()
+
+    def configure_board(self):
+        self.com, self.board, self.pin_a0, self.pin_a2, self.pin_d9 = startup_routine()
+
 
-# Run the experiment and get the config_data
-board_objects = (board, a0, a2, d9)
-times, voltages, currents = experiment(config_data, *board_objects)
+try:
+    # Initialize Experiment
+    board_instance = BoardCom()
+    # com, board, a0, a2, d9 = startup_routine()
+except KeyboardInterrupt:
+    sys.exit("Connection interrupted. Make sure the potentiostat is plugged in then restart the software.\n"
```

Comment:
maybe make it explicit that this was a user initiated interrupt?

Reply:
Ah, good point.